CMP1903_A1_2324/Game.cs

```diff
@@ -1,69 +1,178 @@
-         *
-         * EXTRA: For extra requirements (these aren't required though), the dice rolls could be managed so that the
-         * rolls could be continous, and the totals and other statistics could be summarised for example.
-         */
-
-        //Methods
-
+        public int result
+        {
+            get { return _result; }
+            set { _result = value; }
+        }
+        public int noOfRolls
+        {
+            get { return _noOfRolls; }
+            set { _noOfRolls = value; }
+        }
+        public List<int> results
+        {
+            get { return _results; }
+            set { _results = value; }
+        }
+        //constructor for the class
+        public Game()
+        {
+            D1 = new Die();
+            D2 = new Die();
+            D3 = new Die();
+            result = 0;
+            results = new List<int>();
+            
+        }
         /*
-         * sumup methods will roll all 3 dice 
-         * it will then add the dice together and return the result
+         * these are the methods for this class
+         * SumUp should addup all 3 die and then store the results inside the the list it will then return it
+         * amountOfRolls allows the user to reroll as many times as the user wants
+         * reRoll rerolls the die from the amount the user has given
+         * addUpAll should add up all the dies roll together
+         * averageDie will take all the die added up together then divide it by the amount the user has asked to reroll
+         * minResults will find the minimum score of what the user got
+         * maxResults will find the max score the user got
+         * displayInformation will display all the data gathered
          */
         public int Sumup()
         {
             //rolls 3 dice
-            int D1Roll = _D1.roll();
-            int D2Roll = _D2.roll();
-            int D3Roll = _D3.roll();
+            int D1Roll = D1.roll();
+            int D2Roll = D2.roll();
+            int D3Roll = D3.roll();
+
             //adds the 3 dice rolls rogether
-            int result = D1Roll + D2Roll + D3Roll;
+            result = D1Roll + D2Roll + D3Roll;
+            //adds the results to the list
+            results.Add(result);
             //returns the results from the method
             return result;
         }
 
-        //These methods will get and return all 3 of the dice objectsbut cannot be edited as it is a private object
-        public Die d1
+        public void AmountOfRolls()
         {
-            get { return _D1; }
+            //asks how much the user wants to roll then takes the users input
+            Console.WriteLine("How many times would you like to reroll");
+            int numberOfRolls = int.Parse(Console.ReadLine());
+            // will set the built in variable to what the user has inputted
+            noOfRolls = numberOfRolls;
```

Comment:
Error not handled correctly, when ambiguous input is given an error occurs causing the program to crash